### ISSUE TITLE
NameError occurs during require unless redis-client is required first

### ISSUE THREAD

**Issue by willnet** [sds/mock_redis#318]
[This commit](https://github.com/sds/mock_redis/commit/615c184c6b18c8e1a1bf199963129c99514ca70d) is likely the cause of the issue. Requiring mock_redis without requiring redis-client now results in the following error. 

```
/Users/willnet/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/mock_redis-0.48.0/lib/mock_redis/assertions.rb:4:in `<class:MockRedis>': uninitialized constant MockRedis::RedisClient (NameError)

  DUMP_TYPES = RedisClient::RESP3::DUMP_TYPES
```

Reproducible Code:

```ruby
require 'bundler/inline'

gemfile do
  # Reordering the gem loading sequence makes it work.
  gem 'mock_redis', '0.48.0'
  gem 'redis-client', '0.22.2'
end
```

**Comment by hieuk09** [comment]
Thanks for reporting the bug, I submitted a PR to fix this.

**Comment by sds** [comment]
Thanks for the report. Fix released in [0.48.1](https://github.com/sds/mock_redis/releases/tag/v0.48.1)
